core/src/consensus/tendermint/worker.rs

```diff
@@ -2121,3 +2113,46 @@ fn calculate_score(height: Height, view: View) -> U256 {
     let height = U256::from(height);
     u256_from_u128(std::u128::MAX) * height - view
 }
+
+struct MutTrigger<T> {
+    target: T,
+    deref_mut_triggered: Cell<bool>,
+}
+
+impl<T> MutTrigger<T> {
+    fn new(target: T) -> Self {
+        Self {
+            target,
+            deref_mut_triggered: Cell::new(true),
+        }
+    }
+
+    fn borrow_if_mutated(&self) -> Option<&T> {
```

Comment:
How about adding some words like "reset" in the method name?

Reply:
It is actually `borrow_if_the_target_is_possibly_mutated_after_last_borrow`. It is too long and even longer than the body.

Reply:
I think it would be better to find a good name for the type rather than the method.